doc/decisions/0_drafts/keyset_types.md

```diff
@@ -0,0 +1,79 @@
+# Types of `KeySet`s
+
+## Problem
+
+A `KeySet` can be used in different ways.
+Among these are at least:
+
+- As originally intended, a set of configuration data
+- Metadata of a `Key`
+- General associative array data structure
```

Comment:
This would be perfect to have in use cases.

Reply:
https://github.com/ElektraInitiative/libelektra/blob/3e9036fc91533cc03dfbb6c3fc0a01cf45f12788/doc/usecases/core/README.md#L5-L12